### ISSUE TITLE
Nav Unification:  Not enough screen space for some themes / plugins

### ISSUE THREAD

**Issue by davipontesblog** [Automattic/wp-calypso#50765]
The starter templates screen in WP Admin gets cut on the left - the content bleeds behind the Nav sidebar.

<img width="1272" alt="Screen Shot 2021-03-04 at 12 24 03 PM" src="https://user-images.githubusercontent.com/31009008/109996533-b3278a00-7ce5-11eb-8ed7-73ce6cfd6541.png">

### Steps to reproduce the behavior

1. Install Astra theme on the site. Free version.
2. Go to My Site > Appearance > Astra Options and install the Starter Templates plugin
3. Go to My Site > Appearance > Starter Templates, select any option, like Gutenberg for example, and select a template.
4. Verify the screen gets cut on the left.

**Comment by cpapazoglou** [comment]
The problem here is that some plugins are using `position: fixed` with `160px` sidebar width in mind. We will probably need to find a good solution to that problem.

![](https://cln.sh/ZvoxDR+)

**Comment by getdave** [comment]
There is a clear rationale for opening a discussion about reverting the Nav Unification design sidebar width to match Core. There is no possible way we can add edge cases for the whole Plugin/Theme repo all of which will assume the width supplied in the Core nav.

**Comment by cpapazoglou** [comment]
It seems that we are going to have a problem with an unexpected number of plugins. For history we already had problems with Jetpack and Gutenberg

https://github.com/Automattic/jetpack/blob/16fd659047e04f94b8f3fb768dd15f665eee1291/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.css#L29-L49

I am also in favor of reverting both Calypso and WP Admin to `160px` width. In that case, we would need to fix the following:
1. Site card, title and domain are usually longer than `160px`
2. Nudges & Notices
3. Stats graph, probably a resize (and less vertical bars - data points) will be enough

![](https://cln.sh/apAs75+)

cc @sfougnier

**Comment by cpapazoglou** [comment]
FWIW, here is the issue in which we decided to go from `200px` to `272px`
https://github.com/Automattic/wp-calypso/issues/46912

**Comment by getdave** [comment]
@sfougnier Please could you let us know your thoughts?

**Comment by obenland** [comment]
How does Astra handle collapsed admin menus?

Has this been the only report of the sidebar width breaking things or have there been more? Before we start re-examining a fairly fundamental decision from early with a project, I think we should be sure that it's causing unfixable headaches in more than one theme.

**Comment by ghost** [comment]
>How does Astra handle collapsed admin menus?
>Has this been the only report of the sidebar width breaking things or have there been more?

+1 to these questions.

I would like to avoid reverting back to the original width. It seems a little short-sighted and will impact multiple things (ex: other components in the sidebar, the relationship of the navigation to the grid width of the rest of the screens).

**Comment by cpapazoglou** [comment]
> How does Astra handle collapsed admin menus?

Let me say that the problem is not with the Astra theme but with Astra Starter Templates plugin. With collapsed sidebar it looks OK

![](https://cln.sh/hDmTTv+)

Are you suggesting collapsing the sidebar in certain cases?

> Has this been the only report of the sidebar width breaking things or have there been more? Before we start re-examining a fairly fundamental decision from early with a project, I think we should be sure that it's causing unfixable headaches in more than one theme.

We already had problems with Gutenberg and Jetpack which we fixed by overwriting styles
https://github.com/Automattic/jetpack/blob/16fd659047e04f94b8f3fb768dd15f665eee1291/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.css#L29-L49

**Comment by obenland** [comment]
> Are you suggesting collapsing the sidebar in certain cases?

No I was just wondering if it's maybe conflicting with more than that.

I'll try and reach out to Brainstorm Force and see if they can fix it in the plugin.

**Comment by getdave** [comment]
Reading the comments above, I'm not convinced we have enough information to make a call either way here.

As a result, I'm keen we remain open to the possibility that we may need to revert to match Core to ensure better compatibility with 3rd party code which _expects_ Core not WPCom. I think it's important we aren't coupling ourselves to previous decisions.

How about we ask ~@davipontesblog~ Happiness (with help) to compile an ongoing list of Themes/Plugins which exhibit Issues with the sidebar. These should include a reference to determine whether they are in the Top 10 Themes/Plugins used on WP.com.

Once either:

1. A month has passed, _OR_...
2. Happiness (@mrfoxtalbot) notifies us that the position is becoming untenable (ie: they reach a threshold where HEs feel they are being overwhelmed with pings about "broken" Themes/Plugins.

...we undertake a review to determine how we should proceed.

The options available to us are probably:

1. Do nothing and tell customers they have to accept it because we need a wider sidebar on WP.com for visual/presentational reasons.
2. Undertake a review of the top 10 Themes/Plugins on WordPress.com and ensure they are all compatible with the wider sidebar. Any that aren't we apply "fixes" to in _our_ codebase or (if we have good relations) ask them to fix. Please note this option will incur a maintenance overhead as Themes/Plugins will change over time which could break our fixes without notice.
3. Redesign the sidebar to match Core. This is the best option for maintainability and compatibility but it has downsides in terms of our product visuals. However, we should remember we wanted to align with Core and this is the only area where we have deviated.

Hopefully, an evidence and data-driven approach will ensure we're following the best route for our customers.

@obenland would you be able to make a call on whether you'd be happy for us to proceed with the approach outlined above?

**Comment by davipontesblog** [comment]
@getdave on our end please liaison with @mrfoxtalbot on this, similarly to how we are tracking plugin and theme issues with nav menu items not showing, it seems to be working well on that front.

**Comment by getdave** [comment]
@davipontesblog Yep whoever on Happiness is best placed to keep track. @mrfoxtalbot has been doing a great job!

**Comment by cpapazoglou** [comment]
In our recent Backlog meeting AJAX decided that this is a Blocker for ramping up nav-unification _till we have a solid decision / direction._ 

### Why
Showing to users broken interfaces results to bad user experience. Since the affected screens are somehow usable, users will probably not contact support but they will still be disappointed. There might also be cases where the broken interface will not let users complete an action at all. Ramping up will not show us whether we do have a problem rather than how much **users are annoyed so that they contact us.**

### Do we really have a problem?
As exhibited in https://github.com/Automattic/wp-calypso/issues/50765#issuecomment-796846291, **Jetpack** and **Gutenberg** already take for granted a `160px` sidebar width. Let's not forget that these plugins are developed with best practices in mind. **Astra starter templates** just helped us realize that adding overrides to our [admin-menu.css](https://github.com/Automattic/jetpack/blob/16fd659047e04f94b8f3fb768dd15f665eee1291/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.css#L29-L49) will not solve the problem, it will just create a maintenance burden and a lot more technical debt.

### What can we do now?
We understand that ramping up nav-unification is very important and we can't block it till we have a proper solution (more on this below), that's why we came up with the following short term solutions **for third-party plugins screens only**:

- force `160px` sidebar width.
- auto-collapse the sidebar and switch back to user preference in other screens.
- instead of auto-collapsing the sidebar, we can explore hiding the sidebar and showing an X button (like the customizer).

### Long term solutions
1. Redesigning the sidebar to match Core. This is the best option for maintainability and compatibility but it has downsides in terms of our product visuals. **However, we should remember we wanted to align with Core and this is the only area where we have deviated.**
2. Supporting variable sidebar width in Core first so that all plugins / themes have a chance to adapt.

**Comment by obenland** [comment]
I'd like to propose a two-sided approach, where we increase to 20% as-is and see if we run into more reports (cc @mrfoxtalbot) and give @sfougnier a chance to prepare possible alternatives for a 160px menu.

The upgrade nudges drive a considerable amount of traffic and I worry about negative effects on that with a narrower sidebar.  I agree that maintaining fixes for more plugins would be cumbersome, so let's give ourselves a chance to better understand how many exceptions we're actually talking about before making that call. If it's five, it might be worth the time in exchange for the benefit we maintain.

With Saul working on alternatives, we'll bring ourselves into a better position to make _informed_ changes if we end up deciding to go with the narrow sidebar. Beyond that, we could continue the discussion on P2 and see if narrowing the sidebar would be an enhancement beyond just the maintenance work that incompatible plugins would introduce.

/cc @davemart-in @4thhubbard @lcollette @olaolusoga

**Comment by davemart-in** [comment]
I don't see evidence of this being a widespread issue.

From my perspective we're just kicking the can down the road if we decide to revert to 160px. When the core left nav is finally updated, I don't foresee it sticking with the current 160px width. We're going to have to address this problem sooner or later.

Rather than reverting, can we come up with a way to indicate somehow in the DOM what the width of the left nav is (without needing to use JS) and suggest that plugins update their code to accommodate the third option (menu widths outside of 160px and the collapsed menu bar)?

**Comment by mrfoxtalbot** [comment]
I wanted to share a new report (pc4f5j-pf-p2#comment-869 via @ivan-ottinger) of a similar problem on a different plugin, https://wordpress.org/plugins/folders/

This is how the plugin should work:
![folders](https://user-images.githubusercontent.com/4452464/112027281-5296aa80-8b37-11eb-828c-2772f3b093a4.gif)


... and this is what the plugin looks like with the Unified Nav

<img width="1221" alt="Screenshot 2021-03-22 at 17 53 36" src="https://user-images.githubusercontent.com/4452464/112027568-9689af80-8b37-11eb-9d60-f5eb6251e012.png">


Would we still consider this a bad practice in the way the plugin is coded?

**Comment by cpapazoglou** [comment]
> I don't see evidence of this being a widespread issue.
> 
> From my perspective we're just kicking the can down the road if we decide to revert to 160px. When the core left nav is finally updated, I don't foresee it sticking with the current 160px width. We're going to have to address this problem sooner or later.
> 
> Rather than reverting, can we come up with a way to indicate somehow in the DOM what the width of the left nav is (without needing to use JS) and suggest that plugins update their code to accommodate the third option (menu widths outside of 160px and the collapsed menu bar)?

If we are going to suggest to plugin developers to make a change, let's propose to **them** that they use JS to get the sidebar width which is something they should have done in the first place IMO. We could also think about using a CSS Variable but it won't be IE11 compatible.

**Comment by obenland** [comment]
> let's propose to them that they use JS to get the sidebar width which is something they should have done in the first place IMO.

We're modifying admin menu in quite unexpected ways here, I think it's reasonable to assume the menu width is constant. Can we maybe offer them a patch here that would fix the issue?

**Comment by mrfoxtalbot** [comment]
I just referenced a related issue (https://github.com/Automattic/wp-calypso/issues/51388). 

To be blunt, I am not sure what the added benefit of having a wider Navigation is, while the drawbacks seem to be, albeit not severe, quite a few. I consider reducing the width to match that of the core one.

**Comment by tjcafferkey** [comment]
There is an on-going discussion at the following link trying to come up with a suitable solution for this. pbAPfg-1tk-p2

**Comment by obenland** [comment]
Brainstorm Force has added compatibility with the wider nav in their Starter Templates plugin and will push out an update soon

**Comment by cpapazoglou** [comment]
> I wanted to share a new report (pc4f5j-pf-p2#comment-869 via @ivan-ottinger) of a similar problem on a different plugin, https://wordpress.org/plugins/folders/
> 
> This is how the plugin should work:
> ![folders](https://user-images.githubusercontent.com/4452464/112027281-5296aa80-8b37-11eb-828c-2772f3b093a4.gif)
> 
> ... and this is what the plugin looks like with the Unified Nav
> 
> <img alt="Screenshot 2021-03-22 at 17 53 36" width="1221" src="https://user-images.githubusercontent.com/4452464/112027568-9689af80-8b37-11eb-9d60-f5eb6251e012.png">
> 
> Would we still consider this a bad practice in the way the plugin is coded?

@mrfoxtalbot since we rolled out to 100% are there any other reports besides https://wordpress.org/plugins/folders/ ?

**Comment by mrfoxtalbot** [comment]
Nothing new yet, but I have a dozen reports I will report between today and tomorrow. If I spot something related, I will add it here.

**Comment by kwight** [comment]
This feel like it'll be a long-standing issue, without everyone even agreeing on the severity or where the fixes should happen; I'm dropping this to `normal` while we come up with a more cohesive plan.

**Comment by obenland** [comment]
@cpapazoglou @mrfoxtalbot Other than the Folders plugin, are there any other conflicting plugins that we're aware of?

**Comment by mrfoxtalbot** [comment]
> @cpapazoglou @mrfoxtalbot Other than the Folders plugin, are there any other conflicting plugins that we're aware of?

I am not aware of any other examples, @obenland.

**Comment by cpapazoglou** [comment]
@obenland I have spotted a UI glitch with **All in One SEO**

![](https://cln.sh/Ub8r10+)

**Comment by mrfoxtalbot** [comment]
I have spotted something related affecting **Updraft Plus:**

![Screen Shot on 2021-04-16 at 16:32:45](https://user-images.githubusercontent.com/4452464/115040234-81533700-9ed1-11eb-9bad-b141b982c007.png)

This is what it looks like with the "regular " navigation:

![Screen Shot on 2021-04-16 at 16:32:24](https://user-images.githubusercontent.com/4452464/115040300-929c4380-9ed1-11eb-8343-82eaba51923e.png)

**Comment by liviopv** [comment]
I spotted this issue with the [JobCareers theme](https://themeforest.net/item/jobcareer-job-board-responsive-wordpress-theme/14221636). The theme applies a fixed width of 160px to `#wpcontent` and `#wpfooter`, which overrides the style added by Jetpack.

I added this PHP snippet as a fix via Code Snippets:

```
add_action('admin_head', 'nav_uni_fix');

function nav_uni_fix() {
  echo '<style>
    #wpcontent, #wpfooter {
    margin-left: 272px;
    } 
  </style>';
}
```

3947858-zen

**Comment by mrfoxtalbot** [comment]
Possibly related https://github.com/Automattic/wp-calypso/issues/52214

**Comment by shamserSuzon** [comment]
I have found a UI glitch in the Store Admin.

**Summary of Bug:** There is a visibility/CSS issue in the Store Admin's "My Domains" section.

**Steps to Reproduce:** Go to Store Admin >> Search a user by Username/Email >> Click the "My Domains" button beside the domain name >> My Domains page will appear and now click the "Edit Domain" button >> A pop-up will appear regarding that specific domain's detail information.


![Screen Shot on 2021-11-15 at 16:57:36](https://user-images.githubusercontent.com/4452464/141813902-88e4ab3a-96ef-4cb1-8d18-f0bed86bf1ab.png)

![Edit Domain](https://user-images.githubusercontent.com/53257625/122767056-0e48b100-d2c4-11eb-840a-1410c3425f67.png)


**Estimated Result:** The pop-up window should be displayed above the left side admin menu bar.

**Actual Result:** The pop-up is displaying behind the left side admin menu bar. The pop-up window's contents are getting cut off.

![Screenshot_5](https://user-images.githubusercontent.com/53257625/122767082-17398280-d2c4-11eb-9a70-5dc32e84ffb1.png)


Thank you.

**Comment by davipontesblog** [comment]
Changed the priority to "High".  This is now affecting the core `Appearance > Widgets` for WoA sites.

**Comment by Copons** [comment]
This might be related to https://github.com/Automattic/wp-calypso/issues/51388.

**Comment by mrfoxtalbot** [comment]
> Changed the priority to "High". This is now affecting the core Appearance > Widgets for WoA sites.

I have not been able to reproduce this. Widgets seem to be working as expected. 

Below is a screencast of what I see on my AT testing site at
https://mrfxvanillaat.wpcomstaging.com/wp-admin/widgets.php

https://user-images.githubusercontent.com/4452464/141812102-57a2881b-6e0c-4234-a863-8ff2c734fc4a.mp4

Are you seeing something different, @davipontesblog?

**Comment by davipontesblog** [comment]
Hi @mrfoxtalbot , thanks for testing, and my apologies for not being more specific [here in my comment](https://github.com/Automattic/wp-calypso/issues/50765#issuecomment-967157445).

The issue lies in when adding a Paragraph block as a widget, then trying to add a Link to the word, the link modal from the block toolbar bleeds under the nav bar on the left. I can't share the video I have on this here due to customer privacy, and funnily enough I can't see "Widgets" under Appearance on my WoA sites to test now, not sure why.

**Comment by davipontesblog** [comment]
I was able to replicate the issue on Simple sites just now, from Appearance > Widgets (wp-admin/widgets.php)

https://user-images.githubusercontent.com/31009008/141828517-cca78963-ba26-4614-b4f0-84a7ce6e30f2.mp4

**Comment by Copons** [comment]
Thanks for the clarification @davipontesblog, admittedly I'd have never thought of testing the URL popover! 😅 

This though led me to uncover another, worse issue: the Widgets screen has the editor moved by 272px to the left to make room for the sidebar, but it doesn't take into account the fact that the sidebar can also be collapsed manually:

| Expanded | Collapsed |
| - | - |
| <img width="764" alt="Screenshot 2021-11-15 at 18 09 42" src="https://user-images.githubusercontent.com/2070010/141832652-52d01311-b266-4d3a-b500-a9a42f754310.png"> | <img width="753" alt="Screenshot 2021-11-15 at 18 09 58" src="https://user-images.githubusercontent.com/2070010/141832670-c36da5f4-8f1f-4ed6-aa39-9e590987e0cd.png"> |

Still, I'm not quite sure why the URL popover is rendered outside its intended boundaries. 🤔

**Comment by BogdanUngureanu** [comment]
@davipontesblog I've managed to reproduce this issue with nav-unification disabled. Looks like it might be a Gutenberg issue; therefore, I've created a separate issue for it: https://github.com/WordPress/gutenberg/issues/36953

@shamserSuzon Since the issue is not related with third-party plugins/themes, I've created a separate issue for this based in your screenshots: https://github.com/Automattic/wp-calypso/issues/58572 thanks for reporting it. 

Since these two bugs are now tracked into separate issues, I'll change the priority of this one back to normal.

**Comment by mmtr** [comment]
Now that https://github.com/Automattic/wp-calypso/issues/58572 has been fixed, and after confirming that the [URL popover issue](https://github.com/WordPress/gutenberg/issues/36953) is not caused by Nav Unification, I don't know what else we can do to improve the compatibility of our custom sidebar width with 3rd-party plugins/themes.

To recap, so far we've found conflicting styles in these plugins/themes: [Folders](https://wordpress.org/plugins/folders/), [All in One SEO](https://wordpress.org/plugins/all-in-one-seo-pack/), [UpdraftPlus](https://wordpress.org/plugins/updraftplus/), and [JobCareer](https://themeforest.net/item/jobcareer-job-board-responsive-wordpress-theme/14221636).

It's been suggested to propose fixes to the authors of these plugins/themes so they can support our custom sidebar, but that feels unfair to me. I fully agree with what @tjcafferkey mentioned in pbAPfg-1tk-p2#comment-2936:

> That being said, my thoughts around putting the onus on plugin developers to fix the problem that we have introduced are that it feels unfair in some respects. As far as they are concerned their plugin or theme is compatible with WordPress, but not entirely compatible with WordPress.com since the introduction of a feature that is specific to WPCOM (Nav Unification)

If we really want to fix the Nav Unification incompatibilities with those plugins/themes, I think it's responsibility of the Nav Unification feature to solve them. We can maintain a list of CSS overrides like [this one](https://github.com/Automattic/wp-calypso/issues/50765#issuecomment-829333766) to prevent all the known conflicting styles. However, the cost of maintaining these overrides (they can get easily broken every time a plugin/theme is updated) doesn't seem to compensate the benefit of preventing the issue (all the conflicting styles identified so far minimally affect the UX).

In the long term, Nav Unification should adopt the menu width defined by Core so there are no style inconsistencies in the menu. The main blocker for this to happen is that right now the Core's menu width (160px) is too narrow to support well our upsell nudges. Good news is that Core is exploring to revamp the admin menu with a wider width (https://core.trac.wordpress.org/ticket/47012, https://core.trac.wordpress.org/ticket/50539). So once those changes are in place, it should be relatively straightforward to match the width of the unified sidebar to Core.

Until then, my recommendation is to close this issue out, and if we identify a fully broken UX flow in any 3rd party plugin/theme, fix that with a custom CSS override implemented in Nav Unification itself.

Thoughts?

**Comment by Copons** [comment]
I agree that we shouldn't ask plugin developers to fix an issue introduced by us going off standard.

By the look of this, the problem happens with absolute/fixed elements such as banners or modals, that are typically positioned manually against the Core sidebar, which is traditionally complicated to handle (it has multiple different widths and collapsed states, triggered at different breakpoints by both CSS and JS).

We can debate whether it's an appropriate practice, but it's undeniable that plugin developers have Core in mind and cannot be expected to address a compatibility issue with our business plan — just as much as we can't be expected to manually fix all incompatible plugins.

This leaves us with three options:

- Update the unified sidebar to have the same width as the Core sidebar (including all the responsive quirks).
We have business reasons against this because it'd be too narrow for our nudges, but we could revisit their design to make them more vertical (e.g. move the CTA below the message).
- Hang tight and hope the Core sidebar will become larger soon enough.
- Just... wait.

Looking at the affected plugins:

- JobCareer: not sure about the impact, but it's a premium theme with ~8000 purchases.
- Folders: clear UX issue, but it's a plugin with ~30000 active installations.
- AIO SEO and Updraft: both with ~3 million active installations, but their UX is minimally affected.

So far the impact of this bug is low enough that I think we could delay taking a decision, and do what @mmtr suggested.

If HEs will start receiving too many requests on this issue — and this hasn't happened in many months after the 100% rollout of the unified sidebar — we can triage and manually fix incompatibilities with major plugins.
